packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -259,6 +335,23 @@ class MasterDetailLayout extends ResizeMixin(ElementMixin(ThemableMixin(PolylitM
       this.setAttribute('overlay', '');
     }
   }
+
+  /** @private */
+  __detectVerticalMode() {
```

Comment:
I noticed some issues:
- When setting a master min size in the dev page, the overlay "flickers", so the mode switches with every resize.
- When setting a detail min size, a master size and max-height in the dev page, the overlay does not revert back if there is enough space again.

Reply:
Thanks, will take a look. I also noticed flickering with setting `masterMinSize` to 100% due to scrollbar.

Reply:
The workaround with setting `position: static` is not needed when `masterMinSize` is set.
Updated the PR, now the first issue with flickering should be fixed.

Regarding the second one: it was caused by the border on the detail area on the dev page. Changed to use `clientHeight` instead of `offsetHeight` so that calculation doesn't include borders.

Reply:
With the new changes, there is an issue where it doesn't remove the overlay if you only set a master min size and then increase the size of the layout again.

Reply:
Updated to remove `overlay` attribute first and then only set it back when necessary, this seems to not have this problem (not sure how that would work with animations though).

Reply:
Let's go with this for now, we can refine this later.